tools/build/android/mrwebrtc-unityplugin/build.gradle

```
@@ -0,0 +1,69 @@
+apply plugin: 'com.android.library'
+
+android {
+    compileSdkVersion rootProject.ext.compileSdkVersion
+    buildToolsVersion rootProject.ext.buildToolsVersion
+    defaultConfig {
+        project.archivesBaseName = "mrwebrtc-unityplugin"
+        minSdkVersion rootProject.ext.minSdkVersion
+        targetSdkVersion rootProject.ext.targetSdkVersion
+        versionCode rootProject.ext.versionCode
+        versionName rootProject.ext.versionName
+        ndk {
+            abiFilters "arm64-v8a"
+        }
+        packagingOptions {
+            doNotStrip "*/arm64-v8a/*.so"
+        }
+        externalNativeBuild {
+            cmake {
+                cppFlags "-std=c++17 -DANDROID_STL=c++_shared -DMRS_USE_STR_WRAPPER -DMR_SHARING_ANDROID -DMR_UNITYPLUGIN_SHARING_ANDROID -DUNITY_ANDROID -DSUPPORT_OPENGL_ES=1"
```

```suggestion
                cppFlags "-std=c++17 -DANDROID_STL=c++_shared  -DMR_UNITYPLUGIN_SHARING_ANDROID -DUNITY_ANDROID -DSUPPORT_OPENGL_ES=1"
```
`MRS_USE_STR_WRAPPER` is obsolete, can be removed. `MR_SHARING_ANDROID` should only be used when building mrwebrtc as from previous comment.

---

I removed MRS_USE_STR_WRAPPER but cannot remove MR_SHARING_ANDROID because the unityplugin references files in mrwebrtc that need that define.

---

Right, it seems that some code is using MR_SHARING_ANDROID as a generic "is this Android" check rather than as "are we building mrwebrtc for Android". Will log an issue, but this can go ahead as long as it works.

---

Logged as #629.